SixTrack/sixtrack.s

```diff
@@ -62993,6 +63435,270 @@ c$$$     &           myalphay * cos(phiy))
+!     PARID  - Parent ID
+!     WEIGH  - Statistical Weight
+!     X      - Horizontal coordinate [m]
+!     Y      - Vertical coordinate [m]
+!     Z      - Longitudinal coordinate [m]
+!     XP     - dX/ds
+!     YP     - dY/ds
+!     ZP     - dZ/ds
+!     A      - Nuclear mass number
+!     Z      - Nuclear charge number
+!     m      - Particle mass [GeV/c^2]
+!     P      - Particle momentum [GeV/c^2]      
+!
+      subroutine dist_read( napx, npart, enom, pnom, clight,
+     &     x, y, xp, yp, s, pc, aa, zz, m )
+!
+!
+      implicit none
+
+      
++ca inidist
+!+ca dist_var
+
+
+      
+      character*80   filename
+
+!     interface variables:
+      integer napx, npart
+      double precision enom, pnom, clight
+      double precision x, y, xp, yp, s, pc
+      !double precision dist_en
+
+      !if(dist_type.eq.'NORM') then
+         double precision xn,yn,xpn,ypn,sn,pcn
+      !end if
+      
+      double precision m0                    ! hisix
+
+     ! double precision betx, bety, alfx, alfy
+      
+!     temporary variables:
+      integer id, gen, aa, zz
+      double precision weight, z, zp, m, dt
+      integer jj
+      character*240 tmp_line
+
+      character comment_char
+      parameter ( comment_char = '*' )
+
+      dimension x ( npart ), y ( npart )
+      dimension xp( npart ), yp( npart )
+      dimension s ( npart ), pc( npart )
+      dimension dt( npart )
+      !dimension dist_en(3)
+
+      !if(dist_type.eq.'NORM') then
+         dimension xn ( npart ), yn ( npart )
+         dimension xpn( npart ), ypn( npart )
+         dimension sn ( npart ), pcn( npart )
+      !endif
+
+      ! P. HERMES hisix
+      dimension aa( npart ), zz( npart )
+      dimension m ( npart )      
+      
+      !double precision enom, pnom
+      !integer napx, npart
+      
+      save
+
+      WRITE(*,*) '--> DIST_READ SUBROUTINE CALLED'
+      WRITE(*,*) '--> DIST_SR filename ', dist_fname
+      WRITE(*,*) '--> DIST_SR e0, p0   ', enom, pnom
+      WRITE(*,*) '--> DIST_EN          ', dist_en
+
+      dist_read_unit = 154
+!      dist_echo_unit = 152
+
+
+      
+!     initialise tracking variables:
+      do jj=1,npart
+        x (jj) = 0.0D+00
+        y (jj) = 0.0D+00
+        xp(jj) = 0.0D+00
+        yp(jj) = 0.0D+00
+        pc(jj) = 0.0D+00
+        s (jj) = 0.0D+00
+	aa(jj) = 0            ! hisix
+	zz(jj) = 0            ! hisix
+	m (jj) = 0.0D+00        ! hisix
+        if(dist_type.eq.'NORM') then
+           xn (jj) = 0.0D+00
+           yn (jj) = 0.0D+00
+           xpn(jj) = 0.0D+00
+           ypn(jj) = 0.0D+00
+           pcn(jj) = 0.0D+00
+           sn (jj) = 0.0D+00           
+        endif
+      enddo
+!     initialise particle counter
+      jj = 0      
+
+
+      write(*,*), '--> DIST, dist_read_unit', dist_read_unit
+      write(*,*), '--> DIST, npart         ', npart
+      write(*,*), '--> DIST, napx          ', napx
+      
+      open( unit=dist_read_unit, file=dist_fname )
```

Comment:
To demand that the file should already exist, add `state=OLD`

Also, check that the unit number is not taken by `inquire`